src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -5784,6 +5784,43 @@ export class StudyViewPageStore
         },
     });
 
+    readonly sampleResourceData = remoteData<{
+        [sampleId: string]: ResourceData[];
+    }>({
+        await: () => [this.resourceDefinitions, this.samples],
+        invoke: () => {
+            const sampleResourceDefinitions = this.resourceDefinitions.result!.filter(
+                d => d.resourceType === 'SAMPLE'
+            );
+            if (!sampleResourceDefinitions.length) {
+                return Promise.resolve({});
+            }
+
+            const samples = this.samples.result!;
+            const ret: { [sampleId: string]: ResourceData[] } = {};
+            const promises = [];
+            for (const sample of samples) {
```

Comment:
@Nelliney there will routinely be many thousands of samples in this list, so i think this code will fire a request for each of them (in fact multipe requests for each resource type).  so we need to find a way to get a summary from backend.  lets discuss at standup 
(fyi @inodb)

Reply:
I spoke @inodb and i now understand there's some limit put on number of items.  @Nelliney does this only make the requests if the user has opened the resource tab?

Reply:
Basically, I agree with you @alisman about improving the performance, it is not the best option to implement this logic on the UI. Moreover, there is another the same problem that might affect the performance due to multiple calls. I spoke with @inodb about this matter. Also, I would like to inform you that the function was taken from PatientViewPageStore (1021 line) where it calls many thousands of samples in this list. 

Yes, it will be used when the tab opens.